util/oidc/oidc.go

```diff
@@ -148,32 +139,99 @@ func (a *ClientApp) oauth2Config(scopes []string) (*oauth2.Config, error) {
 	}, nil
 }
 
+func encrypt(data []byte, passphrase string) ([]byte, error) {
+	hasher := md5.New()
+	hasher.Write([]byte(passphrase))
+
+	block, _ := aes.NewCipher([]byte(hex.EncodeToString(hasher.Sum(nil))))
+	gcm, err := cipher.NewGCM(block)
+	if err != nil {
+		return nil, err
+	}
+	nonce := make([]byte, gcm.NonceSize())
+	if _, err = io.ReadFull(cryptorand.Reader, nonce); err != nil {
+		panic(err.Error())
+	}
+	ciphertext := gcm.Seal(nonce, nonce, data, nil)
+	return ciphertext, nil
+}
+
+func decrypt(data []byte, passphrase string) ([]byte, error) {
+	hasher := md5.New()
+	hasher.Write([]byte(passphrase))
+
+	key := []byte(hex.EncodeToString(hasher.Sum(nil)))
+	block, err := aes.NewCipher(key)
+	if err != nil {
+		return nil, err
+	}
+	gcm, err := cipher.NewGCM(block)
+	if err != nil {
+		return nil, err
+	}
+	nonceSize := gcm.NonceSize()
+	nonce, ciphertext := data[:nonceSize], data[nonceSize:]
```

Comment:
We should make sure that `len(data) > nonceSize` to prevent out-of-bounds access to `data` slice, even if that might be only remotely possible. We should not just assume that `data` will be of the same size to the `nonceSize` or larger.